waku/v2/node/connectedness.go

```diff
@@ -160,20 +161,38 @@ func (w *WakuNode) Status() (isOnline bool, hasHistory bool) {
 			if !hasStore && protocol == store.StoreID_v20beta4 {
 				hasStore = true
 			}
-			if !hasFilter && protocol == legacy_filter.FilterID_v20beta1 {
+			if !hasFilter && protocol == filter.FilterSubscribeID_v20beta1 {
 				hasFilter = true
 			}
 		}
 	}
 
 	if hasStore {
 		hasHistory = true
+	} else {
+		//Look if there are other peers in peerStore that supports Store and we are not connected to them
+		for _, peer := range w.host.Peerstore().Peers() {
+			protocols, err := w.host.Peerstore().GetProtocols(peer)
+			if err != nil {
+				w.log.Warn("reading peer protocols", logging.HostID("peer", peer), zap.Error(err))
+				continue
+			}
+			for _, protocol := range protocols {
+				if !hasStore && protocol == store.StoreID_v20beta4 {
+					hasStore = true
+					break
+				}
+			}
+			if hasStore {
+				break
+			}
+		}
```

Comment:
We're actually not using the logic from line 170 to 189, right?  I'm thinking if we should remove that

Reply:
Also, i think 161-163 is not needed either

Reply:
> We're actually not using the logic from line 170 to 189, right? I'm thinking if we should remove that

I added them thinking that atleast if the peerStore has a storeNode we can indicate it. But yeah, we have on-demand discovery as well so it shouldn't matter i guess.

Reply:
> Also, i think 161-163 is not needed either

Hmm, true. Will remove them